lib/types/src/protocol/execution_version.rs

```diff
@@ -0,0 +1,76 @@
+use num_enum::TryFromPrimitive;
+
+use super::ProtocolSemanticVersion;
+
+/// Identifier of the MultiVM execution version that corresponds to a concrete state transition function.
+/// Generally this is depicted by the minor of the protocol version, e.g. it can (but not guaranteed to) only change
+/// if the minor of the protocol version changes.
```

Comment:
Hmm, really? I assumed execution version could change inside the same protocol version. Imagine we found a critical bug in STF that leads to unprovable block. We then patch zksync-os, release it, add `V5` here and a new corresponding proving version as well. In the meanwhile on L1 level only VKs change and hence protocol version remains the same.

Reply:
Yup, basically the same way as it's done on Era.